pkg/pillar/base/logobjecttypes.go

```diff
@@ -172,13 +176,28 @@ func InitLogObject(logBase *LogObject, object *LogObject, objType LogObjectType,
 }
 
 // NewSourceLogObject : create an object with agentName and agentPid
+// Since there might be multiple calls to this for the same agent
+// we check for an existing one for the agentName
 func NewSourceLogObject(agentName string, agentPid int) *LogObject {
-	object := new(LogObject)
+	// Check if we already have an object with the given agentName
+	var object *LogObject
+	value, ok := logSourceObjectMap.Load(agentName)
+	if ok {
+		object, ok = value.(*LogObject)
+		if ok {
+			return object
+		}
+		log.Fatalf("NewSourceLogObject: Object found is not of type *LogObject, found: %T",
+			value)
+	}
+
+	object = new(LogObject)
 	object.Initialized = true
 	fields := make(map[string]interface{})
 	fields["source"] = agentName
 	fields["pid"] = agentPid
 	object.Fields = fields
+	logSourceObjectMap.Store(agentName, object)
```

Comment:
I suppose the assumption is that these source log objects once created will only be destroyed along with zedbox, right?

Reply:
Yes, because there is currently no way to shut down a zedbox microservice.